features/src/autogluon/features/generators/datetime.py

```diff
@@ -61,16 +65,29 @@ def _generate_features_datetime(self, X: DataFrame) -> DataFrame:
             # TODO: Be aware: When converted to float32 by downstream models, the seconds value will be up to 3 seconds off the true time due to rounding error.
             #  If seconds matter, find a separate way to generate (Possibly subtract smallest datetime from all values).
             # TODO: could also return an extra boolean column is_nan which could provide predictive signal.
-            X_datetime[datetime_feature] = pd.to_datetime(X[datetime_feature], errors='coerce').fillna(self._fillna_map[datetime_feature])
+            # Note: The .replace call is required to handle the obnoxious edge-case of:
+            #   NaN, empty string, datetime without timezone, and datetime with timezone, all as an object type, all being present in the same column.
+            #   I don't know why, but in this specific situation (and not otherwise), NaN will be filled by .fillna, but empty string will be converted to NaT
+            #   and refuses to be filled by .fillna, requiring a dedicated replace call. (NaT is filled by .fillna in every other situation...)
+            X_datetime[datetime_feature] = pd.to_datetime(X[datetime_feature], errors='coerce')\
+                .fillna(self._fillna_map[datetime_feature])\
+                .replace({'NaT': self._fillna_map[datetime_feature]})
             # X_datetime[datetime_feature] = pd.to_timedelta(X_datetime[datetime_feature]).dt.total_seconds()
             # Parse the date into lots of derived fields.
             # Most of the pandas Series.dt properties are here, a few are omitted (e.g. is_month_start) if they can be inferred
             # from other features.
             for feature in self.features:
-                X_datetime[datetime_feature + '.' + feature] = getattr(X_datetime[datetime_feature].dt, feature).astype(int)
-
-            X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
-
+                try:
+                    X_datetime[datetime_feature + '.' + feature] = getattr(X_datetime[datetime_feature].dt, feature).astype(int)
+                except AttributeError:
+                    # Strange datetime object, try removing timezone info
+                    X_datetime[datetime_feature + '.' + feature] = getattr(self._remove_timezones(X_datetime[datetime_feature]).dt, feature).astype(int)
+
+            try:
+                X_datetime[datetime_feature] = pd.to_numeric(X_datetime[datetime_feature])
+            except TypeError:
+                # Strange datetime object, try removing timezone info
```

Comment:
Stripping known time zones would change the timestamp for those values. The method below can handle it correctly.

I think we can explicitly handle edge cases via separating good rows vs bad ones; using `utc=True` allows to normalize all time zones (though the cases without tz would be considered UTC):
```python
def normalize_timeseries(series):
    series = pd.to_datetime(series, utc=True)
    broken_idx = series[(series == 'NaT') | series.isna() | series.isnull()].index
    bad_rows = series.iloc[broken_idx]
    good_rows = series[~series.isin(bad_rows)].astype(int)
    series.iloc[good_rows.index] = good_rows
    series[broken_idx] = int(good_rows.mean())
    return series
```

Reply:
```
series = pd.to_datetime(series, utc=True)
```
Crashes.

```

../../../src/autogluon/features/generators/datetime.py:131: in _remove_timezones
    return normalize_timeseries(datetime_as_object)
../../../src/autogluon/features/generators/datetime.py:122: in normalize_timeseries
    series = pd.to_datetime(series, utc=True)
../../../../../../virtual/autogluon38/lib/python3.8/site-packages/pandas/core/tools/datetimes.py:1068: in to_datetime
    values = convert_listlike(arg._values, format)
../../../../../../virtual/autogluon38/lib/python3.8/site-packages/pandas/core/tools/datetimes.py:438: in _convert_listlike_datetimes
    result, tz_parsed = objects_to_datetime64ns(
../../../../../../virtual/autogluon38/lib/python3.8/site-packages/pandas/core/arrays/datetimes.py:2177: in objects_to_datetime64ns
    result, tz_parsed = tslib.array_to_datetime(
pandas/_libs/tslib.pyx:427: in pandas._libs.tslib.array_to_datetime
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: mixed datetimes and integers in passed array
```